base/abstractarray.jl

```diff
@@ -1143,23 +1140,29 @@ function cat_t(dims, T::Type, X...)
     return _cat(A, shape, catdims, X...)
 end
 
+_cat_inds(i, x, ::Tuple{}, offsets, concat) = ()
```

Comment:
This change is more than is necessary here… and probably has compile-time performance implications.  I reverted it in favor of a simpler work-around.